server/src/api/boards_listen_on_board.go

```diff
@@ -1,153 +1,221 @@
-				err := conn.WriteJSON(filteredMsg)
-				if err != nil {
-					logger.Get().Warnw("failed to send message", "message", filteredMsg, "err", err)
-				}
-			}
-		}
-	}
+func (s *Server) listenOnBoard(boardID uuid.UUID, userSession *dto.BoardSession, conn *websocket.Conn, initEventData EventData, isMod bool) {
+  if _, exist := s.boardSubscriptions[boardID]; !exist {
+    s.boardSubscriptions[boardID] = &BoardSubscription{
+      clients: make(map[realtime.SessionChannel]map[uuid.UUID]*websocket.Conn),
+    }
+    s.boardSubscriptions[boardID].clients[realtime.SessionChannelModerator] = make(map[uuid.UUID]*websocket.Conn)
+    s.boardSubscriptions[boardID].clients[realtime.SessionChannelParticipant] = make(map[uuid.UUID]*websocket.Conn)
+    s.boardSubscriptions[boardID].subscription = make(map[realtime.SessionChannel]chan *realtime.BoardEvent)
+  }
+
+  b := s.boardSubscriptions[boardID]
+  if isMod {
+    b.clients[realtime.SessionChannelModerator][userSession.User.ID] = conn
+  } else {
+    b.clients[realtime.SessionChannelParticipant][userSession.User.ID] = conn
+  }
+
+  b.boardParticipants = initEventData.Sessions
+  b.boardSettings = initEventData.Board
+  b.boardColumns = initEventData.Columns
+  b.boardNotes = initEventData.Notes
+  b.boardReactions = initEventData.Reactions
+
+  if b.subscription[realtime.SessionChannelModerator] == nil {
+    b.subscription[realtime.SessionChannelModerator] = s.realtime.GetBoardChannel(boardID, realtime.SessionChannelModerator)
+    go b.startListeningOnBoard(realtime.SessionChannelModerator)
+  }
+
+  if b.subscription[realtime.SessionChannelParticipant] == nil {
+    b.subscription[realtime.SessionChannelParticipant] = s.realtime.GetBoardChannel(boardID, realtime.SessionChannelParticipant)
+    go b.startListeningOnBoard(realtime.SessionChannelParticipant)
+  }
+
+}
+
+func (b *BoardSubscription) startListeningOnBoard(channel realtime.SessionChannel) {
+
+  for msg := range b.subscription[channel] {
+    targetedSession := parseBoardSession(msg.Data)
+    logger.Get().Debugw("message received", "message", msg)
+    clients := b.clients[channel]
+
+    for id, conn := range clients {
+      filteredMsg := b.eventFilter(msg, id)
+      if msg.Type == realtime.BoardEventParticipantUpdated && id == targetedSession.User.ID {
+        b.changeChannel(channel, targetedSession)
+      }
+      b.Lock()
+      err := conn.WriteJSON(filteredMsg)
+      b.Unlock()
+      if err != nil {
+        logger.Get().Warnw("failed to send message", "message", filteredMsg, "err", err)
+      }
+    }
+
+  }
+}
+
+func (b *BoardSubscription) changeChannel(channel realtime.SessionChannel, oldSession *dto.BoardSession) {
+  if oldSession.Role == types.SessionRoleModerator && channel == realtime.SessionChannelModerator || oldSession.Role == types.SessionRoleParticipant && channel == realtime.SessionChannelParticipant || oldSession.Role == types.SessionRoleOwner && channel == realtime.SessionChannelModerator {
```

Comment:
`oldSession.Role == types.SessionRoleOwner` 
As of know, Owners will always be Owners and they can't change their role